File: src/main/java/com/distribution/modules/integral/service/impl/ProductDetailServiceImpl.java
package com.distribution.modules.integral.service.impl;

import com.distribution.modules.integral.dao.ProductDetailDao;
import com.distribution.modules.integral.entity.ProductDetailEntity;
import com.distribution.modules.integral.service.ProductDetailService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Map;


@Service("productDetailService")
public class ProductDetailServiceImpl implements ProductDetailService {
    @Autowired
    private ProductDetailDao productDetailDao;

    @Override
    public ProductDetailEntity queryObject(String id) {
        return productDetailDao.queryObject(id);
    }

    @Override
    public List<ProductDetailEntity> queryList(Map<String, Object> map) {
        return productDetailDao.queryList(map);
    }


    @Override
    @Transactional(rollbackFor = Exception.class)
    public void save(ProductDetailEntity productDetail) throws Exception {
        productDetailDao.save(productDetail);
    }

    @Override
    @Transactional(rollbackFor = Exception.class)
    public void update(ProductDetailEntity productDetail) throws Exception {
        productDetailDao.update(productDetail);
    }

    @Override
    @Transactional(rollbackFor = Exception.class)
    public void delete(String id) throws Exception {
        productDetailDao.delete(id);
    }

    @Override
    @Transactional(rollbackFor = Exception.class)
    public void deleteBatch(String[] ids) throws Exception {
        productDetailDao.deleteBatch(ids);
    }

}

File: src/main/kotlin/com/distribution/modules/integral/service/impl/ProductDetailServiceImpl.kt
package com.distribution.modules.integral.service.impl

import com.distribution.modules.integral.dao.ProductDetailDao
import com.distribution.modules.integral.entity.ProductDetailEntity
import com.distribution.modules.integral.service.ProductDetailService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional


@Service("productDetailService")
class ProductDetailServiceImpl : ProductDetailService {
    @Autowired
    private lateinit var productDetailDao: ProductDetailDao

    override fun queryObject(id: String): ProductDetailEntity {
        return productDetailDao.queryObject(id)
    }

    override fun queryList(map: Map<String, Any>): List<ProductDetailEntity> {
        return productDetailDao.queryList(map)
    }


    @Transactional(rollbackFor = [(Exception::class)])
    @Throws(Exception::class)
    override fun save(productDetail: ProductDetailEntity) {
        productDetailDao.save(productDetail)
    }

    @Transactional(rollbackFor = [(Exception::class)])
    @Throws(Exception::class)
    override fun update(productDetail: ProductDetailEntity) {
        productDetailDao.update(productDetail)
    }

    @Transactional(rollbackFor = [(Exception::class)])
    @Throws(Exception::class)
    override fun delete(id: String) {
        productDetailDao.delete(id)
    }

    @Transactional(rollbackFor = [(Exception::class)])
    @Throws(Exception::class)
    override fun deleteBatch(ids: Array<String>) {
        productDetailDao.deleteBatch(ids)
    }

}
